addons/l10n_in/report/account_invoice_report.py

```diff
@@ -272,6 +272,17 @@ def _where(self):
         return """
                 WHERE am.state = 'posted'
                     AND tag_rep_ln.account_tax_report_line_id in (SELECT res_id FROM ir_model_data WHERE module='l10n_in' AND name in ('tax_report_line_igst','tax_report_line_cgst','tax_report_line_sgst','tax_report_line_zero_rated'))
+                    AND (
+                            parent_at.id is NULL
+                            OR
+                            parent_at.id in (
+                              SELECT account_tax_id
+                              FROM account_move_line_account_tax_rel aml_taxes
+                              JOIN account_move_line aml2 on aml_taxes.account_move_line_id = aml2.id
+                             WHERE aml2.move_id = aml.move_id
+                               AND aml2.product_id = aml.product_id
+                            )
```

Comment:
This is very likely to be a perf killer.

Reply:
@william-andre Yes, but right now we don't have any other options.
In future, we refactor the whole GSTR-1 report.

Reply:
> @william-andre Yes, but right now we don't have any other options.
> In future, we refactor the whole GSTR-1 report.

What should we say to a customer who gets timeout error? "Wait 1 year before we refactor the module"?

Reply:
One of the option is creating new module, say `l10n_in_2` that makes required changes without making it unusable in large database

Reply:
@yelizariev Good idea, only needed when child taxes have multi-parent.

Reply:
Having a separate module is better, but it still introduces performance issues. Maybe you can make needed changes via the new module, so it will work without performance issues?

Reply:
@yelizariev No, still it has performance issues.

I find one other way to fix where I read these values.
when the record is duplicated then there is a search result is like
`l10n_in.account.invoice.report(2,2,3,4,4)`
So we can convert this to set and remove duplicates automatically.
what do you say about this?

Note:  this is not happening before this commit https://github.com/odoo/odoo/commit/7a7b5b5d9d01e193c2330fe4fe076c2bf53fbc49#diff-7144f88ea32f36feb17ce1b8dda7dee1631f5ada34075414587df3948c6b3d1bL4377

because we retrun _uniquify_list in ORM.

CC: @william-andre

Reply:
So, the problem is duplicates in search_read ?
```py
                for read_data in self.env[model_domain.get('model')].search_read(domain, model_domain.get('sum_fields')):
```
https://github.com/odoo/enterprise/blob/b4974070aced25b46ad4b1e02165571d01e59a15/l10n_in_reports/report/account_gst_report.py#L126

Reply:
Adding `.union()` at the end will deduplicate it
```diff
-                for read_data in self.env[model_domain.get('model')].search_read(domain, model_domain.get('sum_fields')):
+                for read_data in self.env[model_domain.get('model')].search_read(domain, model_domain.get('sum_fields')).union():
```

Reply:
Anyway, you can deduplicate without changing sql, no?

Reply:
@yelizariev Yes, we can
```
(Pdb) self.env[model_domain.get('model')].search(filter_domain + model_domain.get('domain'))
l10n_in.account.invoice.report(64, 58, 58, 6, 8, 8)
(Pdb) self.env[model_domain.get('model')].search(filter_domain + model_domain.get('domain')).union()
l10n_in.account.invoice.report(64, 58, 6, 8)```

Reply:
@yelizariev and @william-andre 
We can remove duplicates by use union() where I search these records.
So what you suggest we fix it using SQL or union() where we search these records?

Reply:
Better to have a solution, that don't introduce performance issues, i.e. `union()` in this case

Reply:
Note, that if duplicates comes from `search_read` you would need to deduplicate manually (make dictionary or something like that)

Reply:
I do it like `search_read(domain)` to `search(domain).union().read()` 😉